### PR TITLE
Fixed crash when loading a large image as a qr code with settings + refactoring

### LINKED ISSUE #5346: Importing a big sized image while reconfiguring with QR code crashes the app
#### ODK Collect version
v2022.3.0, v2022.4 Beta 2
#### Android version
10, 13 (probably all)
#### Device used
Redmi 9T, Pixel 6a
#### Problem description
The app crashes if a user tries to import a big sized image (which isn’t a QR code) while reconfiguring with QR code.
#### Steps to reproduce the problem
1. Go to Settings, Project management, Reconfigure with QR code
2. Tap on the three dots and import QR code.
3. Select a big sized file (I tried 27MB image).
#### Expected behavior
The app shouldn’t crash. User should see a toast.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5346 

#### What has been done to verify that this works as intended?
I've tested the fix manually.

#### Why is this the best possible solution? Were any other approaches considered?
Apart from the fix which was pretty small I've also:
- converted related classes to kotlin
- moved the code responsible for encoding and decoding qr codes to a separate module called `qr-code`

We could go one step further and move the classes that still exist in configure/qr package to a separate module too but it's not that easy and obvious since they are coupled with settings etc so that would require more work and should be done separatelly.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I've refactored a lot of code responsible for handling qr codes (scanning and creating qr codes) so please test everything in that area carefully.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: androidshared/src/test/java/org/odk/collect/androidshared/utils/CompressionUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.zip.DataFormatException
+
+@RunWith(AndroidJUnit4::class)
+class CompressionUtilsTest {
+    private val text = (
```
Comment (seadowg) [inline androidshared/src/test/java/org/odk/collect/androidshared/utils/CompressionUtilsTest.kt]: The text doesn't need to be as long as this right? I think we can also avoid a shared field here and just define our inputs in each test here.

**Review by seadowg — APPROVED** [review]

File: androidshared/src/test/java/org/odk/collect/androidshared/utils/CompressionUtilsTest.kt (hunk 1 above)
Comment (grzesiek2010) [reply androidshared/src/test/java/org/odk/collect/androidshared/utils/CompressionUtilsTest.kt]: Right, fixed.

**Comment by dbemke** [comment]
#### Android version
8.1, 10, 11, 13
#### Device used
Nexus 5X, Galaxy M12, Redmi 9T, Galaxy A32
#### Problem description
Creating a QR code via Collect with the PR version and then scanning it with an older version of Collect on another device result is "QR code does not contain valid settings” information.
#### Steps to reproduce the problem
1. Add a project in Collect (while on the PR).
2. Go to Settings→ Project Management → Reconfigure with QR code.
3. Tap "QR Code” to create a QR code.
4. Take a different device with previous Collect version (e.g. 2023.1. Beta 0 or 2022.4.4).
5. Scan the QR code from the first device (no matter if it’s adding a project or reconfiguring with QR code).
#### Expected behavior
The QR code should be a valid code.
#### Other information 
I tried:
- scanning a QR code created by an older version of Collect with a device with the PR version – works as expected
- scanning a QR code created by the PR version of Collect with a device with the PR version- works as expected

**Comment by grzesiek2010** [comment]
@dbemke good catch! The problem should be fixed. Please continue testing.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 8.1, 10, 11

**Verified Cases:**
- issue #5346 and the issue mentioned in the PR are no longer reproducing
- regression checks in scanning and reconfiguring with a QR code
- scanning a QR code in the launch screen
- barcodes form
- adding duplicate / switching to an existing project
- Central, Aggregate, Google Drive projects
- creating QR codes via Collect with changed settings, admin/server passwords included and not included in the QR code
- scanning QR codes with an older version of Collect

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Android 12, 13

**Pull request merged**